### PR TITLE
Migrate to the latest Monaco

### PR DESCRIPTION
**Pull request by akosyakov**
Fixes https://github.com/theia-ide/theia/issues/306

### CODE REVIEW

**Comment by akosyakov** [comment]
It still needs more testing

**Comment by akosyakov** [comment]
Fixes #213

**Comment by akosyakov** [comment]
Looks better now :)
palette

**Comment by akosyakov** [comment]
@MarkZ3 Could you test with the cpp extension?

**Comment by MarkZ3** [comment]
I noticed issues but they were present there because this pull request:
- Errors/Warnings not shown when hovering after the context menu is opened once
- document/definition does not jump to correct file offset when opening another file. Always opens new files at 0, 0 (LSP 0-based offset)?
-Quick fixes: not sure how to trigger this. I think it might be a Clangd problem because I see in the log "C/C++: Failed to decode CodeActionParams!"

I will create issues for those.

**Comment by akosyakov** [comment]
@MarkZ3 thank you for the reviewing, I am merging it when the build is green

File: packages/monaco/src/browser/monaco-editor-provider.ts [hunk 1]
```diff
@@ -13,18 +13,22 @@ import { EditorPreferences, EditorPreferenceChange } from "@theia/editor/lib/bro
 import { MonacoEditor } from "./monaco-editor";
 import { MonacoEditorModel } from './monaco-editor-model';
 import { MonacoEditorService } from "./monaco-editor-service";
-import { MonacoModelResolver } from "./monaco-model-resolver";
+import { MonacoTextModelService } from "./monaco-model-resolver";
```
Comment (svenefftinge) [inline packages/monaco/src/browser/monaco-editor-provider.ts]: If we rename the type we should rename the file, too

**Pull request merged**